src/interface/sparse_pack.hpp

```diff
@@ -119,124 +120,26 @@ class SparsePack : public SparsePackBase {
-                const std::vector<MetadataFlag> &flags = {}) {
-    const bool coarse = true;
-    const bool fluxes = false;
-    Get(pmd, vars, flags, fluxes, coarse);
-  }
-
-  template <class T>
-  static SparsePack GetFlat(T *pmd, const std::vector<MetadataFlag> &flags = {}) {
-    const bool coarse = false;
-    const bool fluxes = false;
-    const bool flatten = true;
-    return Get(pmd, flags, fluxes, coarse, flatten);
-  }
-
-  template <class T, class VAR_VEC>
-  static std::tuple<SparsePack, SparsePackIdxMap>
-  GetFlat(T *pmd, const VAR_VEC &vars, const std::vector<MetadataFlag> &flags = {}) {
-    const bool coarse = false;
-    const bool fluxes = false;
-    const bool flatten = true;
-    return Get(pmd, vars, flags, fluxes, coarse, flatten);
-  }
-
-  template <class T>
-  static SparsePack GetFlatWithFluxes(T *pmd,
-                                      const std::vector<MetadataFlag> &flags = {}) {
-    const bool coarse = false;
-    const bool fluxes = true;
-    const bool flatten = true;
-    return Get(pmd, flags, fluxes, coarse, flatten);
-  }
-
-  template <class T, class VAR_VEC>
-  static std::tuple<SparsePack, SparsePackIdxMap>
-  GetFlatWithFluxes(T *pmd, const VAR_VEC &vars,
-                    const std::vector<MetadataFlag> &flags = {}) {
-    const bool coarse = false;
-    const bool fluxes = true;
-    const bool flatten = true;
-    return Get(pmd, vars, flags, fluxes, coarse, flatten);
-  }
-
-  template <class T>
-  static SparsePack GetFlatWithCoarse(T *pmd,
-                                      const std::vector<MetadataFlag> &flags = {}) {
-    const bool coarse = true;
-    const bool fluxes = false;
-    const bool flatten = true;
-    return Get(pmd, flags, fluxes, coarse, flatten);
-  }
-
-  template <class T, class VAR_VEC>
-  static std::tuple<SparsePack, SparsePackIdxMap>
-  GetFlatWithCoarse(T *pmd, const VAR_VEC &vars,
-                    const std::vector<MetadataFlag> &flags = {}) {
-    const bool coarse = true;
-    const bool fluxes = false;
-    const bool flatten = true;
-    return Get(pmd, vars, flags, fluxes, coarse, flatten);
-  }
+  class Descriptor : public impl::PackDescriptor {
+   public:
+    explicit Descriptor(const impl::PackDescriptor &desc_in)
+        : impl::PackDescriptor(desc_in) {}
+
+    // Make a `SparsePack` from variable_name types in the type list Ts..., creating the
+    // pack in `pmd->SparsePackCache` if it doesn't already exist. Variables can be
+    // accessed on device via instance of types in the type list Ts...
+    // The pack will be created and accessible on the device
+    template <class T>
+    SparsePack GetPack(T *pmd) const {
+      return SparsePack(SparsePackBase::GetPack(pmd, *this));
+    }
+
+    SparsePackIdxMap GetMap() const {
+      PARTHENON_REQUIRE(sizeof...(Ts) == 0,
+                        "Should not be getting an IdxMap for a type based pack");
+      return SparsePackBase::GetIdxMap(*this);
+    }
```

Comment:
I think it's fine not to mix type vs string based packs, but I vaguely remember us coming up with a solution while we were talking to mix them unambiguously.

Reply:
Yeah, I think this is actually very easy to relax, it just requires writing a new `MakeDescriptor` that puts together a vector of variable groups from types and a vector of strings.

Reply:
I would say we leave this for another PR.